components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/Types.kt

```diff
@@ -4,16 +4,44 @@
 
 package mozilla.components.service.sync.logins
 
-import mozilla.components.concept.sync.SyncAuthInfo
+import mozilla.components.concept.storage.Login
 
 /**
- * Conversion from a generic AuthInfo type into a type 'logins' lib uses at the interface boundary.
+ * Conversion from a generic [Login] into its richer comrade within the 'logins' lib.
  */
-fun SyncAuthInfo.into(): SyncUnlockInfo {
-    return SyncUnlockInfo(
-        kid = this.kid,
-        fxaAccessToken = this.fxaAccessToken,
-        syncKey = this.syncKey,
-        tokenserverURL = this.tokenServerUrl
+fun Login.toNativeLogin(): mozilla.appservices.logins.Login {
+    return mozilla.appservices.logins.Login(
+        id = this.guid ?: "",
+        username = this.username,
+        password = this.password,
+        hostname = this.origin,
+        formSubmitUrl = this.formActionOrigin,
+        httpRealm = this.httpRealm,
+        timesUsed = this.timesUsed,
+        timeCreated = this.timeCreated,
+        timeLastUsed = this.timeLastUsed,
+        timePasswordChanged = this.timePasswordChanged,
+        usernameField = this.usernameField ?: "",
+        passwordField = this.passwordField ?: ""
     )
 }
+
+/**
+ * Conversion from a "native" login [Login] into its generic comrade.
+ */
+fun mozilla.appservices.logins.Login.toLogin(): Login {
+    return Login(
+        guid = this.id,
+        origin = this.hostname,
+        formActionOrigin = this.formSubmitUrl,
+        httpRealm = this.httpRealm,
+        username = this.username,
+        password = this.password,
+        timesUsed = this.timesUsed,
+        timeCreated = this.timeCreated,
+        timeLastUsed = this.timeLastUsed,
+        timePasswordChanged = this.timePasswordChanged,
+        usernameField = this.usernameField,
```

Comment:
I wonder if empty strings here should end up as null?

Reply:
I think it'll be nice, yeah! Gets us to the spirit of the `String?` type. We do similar checks in the mr2 storage stuff (but, that's cause w/ protobuf passing around `null` isn't a thing).

Reply:
I also agree! Unfortunately, this might need to be something at the longtail of this entire effort as we have to change the Rust side and propagate all the way through the PRs and mainly trying to keep everything relatively unchanged for when the API re-turfing happens after uniffi